apps/shelter-client/src/libs/Chat.tsx

```diff
@@ -15,34 +15,41 @@ interface Message {
   icon: string;
 }
 
-const socket = io(config.gatewayUrl);
-
 const Chat: React.FC = () => {
-  const updateState = (newState: Partial<IState>): void => setState((prevState) => ({ ...prevState, ...newState }));
-  const [state, setState] = useState<IState>({
-    messages: [],
-    newMessage: ''
-  });
+  const [state, setState] = useState<IState>({ messages: [], newMessage: '' });
+  const socketRef = useRef<Socket | null>(null);
 
   useEffect(() => {
-    socket.on('message', (data: Message) => {
-      updateState({ messages: [...state.messages, data] });
+    socketRef.current = io(config.gatewayUrl);
 
+    socketRef.current.on('message', (data: Message) => {
+      setState((prevState) => ({ ...prevState, messages: [...prevState.messages, data] }));
     });
-  }, [state.messages]);
 
-  const handleSendMessage = () => {
+    return () => {
+      socketRef.current?.disconnect();
+    };
+  }, []);
+
+  const handleSendMessage = useCallback(() => {
     if (state.newMessage.trim() !== '') {
-      socket.emit('message', { sender: 'denys', message: state.newMessage, icon: 'user' });
-      updateState({ newMessage: '' });
+      socketRef.current?.emit('message', { sender: 'User', message: state.newMessage, icon: 'user' });
+      setState((prevState) => ({ ...prevState, newMessage: '' }));
```

Comment:
I think, better to use already implemented function updateState as it done all around the project just to simplify progress a little and remove at least one destructuring assignment